code/datums/components/team_monitor.dm

```diff
@@ -0,0 +1,542 @@
+	//Who are we updating for
+	var/mob/updating = null
+	//Distance from center
+	//Probably in pixels or something idk
+	var/distance = 20
+	//Should we display the hud in the firstplace
+	var/hud_visible = TRUE
+	//The attached beacon: Ignore this one
+	var/datum/component/tracking_beacon/attached_beacon
+
+/datum/component/team_monitor/Initialize(frequency_key, frequency, _attached_beacon)
+	var/obj/item/clothing/item = parent
+	if(!istype(item))
+		return COMPONENT_INCOMPATIBLE
+
+	team_freq_key = frequency_key
+	if(frequency != null)
+		team_frequency = "[frequency_key][frequency]"
+	else
+		team_frequency = null
+
+	attached_beacon = _attached_beacon
+
+	RegisterSignal(parent, COMSIG_ITEM_EQUIPPED, .proc/parent_equipped)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/parent_dequpped)
+
+	get_active_trackers()
+	add_tracker_hud(team_frequency, src)
+
+/datum/component/team_monitor/Destroy(force, silent)
+	//Unregister signals
+	if(parent)
+		UnregisterSignal(parent, COMSIG_ITEM_EQUIPPED)
+		UnregisterSignal(parent, COMSIG_ITEM_DROPPED)
+
+	if(team_frequency)
+		GLOB.tracker_huds[team_frequency] -= src
+
+	//Stop processing
+	STOP_PROCESSING(SSprocessing, src)
+
+	. = ..()
+
+//Gets the active trackers for when the team_monitor component
+//is initialized while other trackers are already active.
+/datum/component/team_monitor/proc/get_active_trackers()
+	for(var/datum/component/tracking_beacon/beacon as() in get_all_beacons_on_frequency(team_frequency, team_freq_key))
+		if(beacon != attached_beacon && (beacon.updating || beacon.always_update))
+			add_to_tracking_network(beacon)
+
+//===========
+// Handles the parent being moved and updates the direction of the arrows.
+//===========
+
+/datum/component/team_monitor/process()
+	update_all_directions()
+
+//When the parent is removed, we need to update our arrows
+//Also if we are visible update the arrows of anything tracking us
+/datum/component/team_monitor/proc/parent_moved()
+	//Update our alt appearences
+	update_all_directions()
+
+//Updates the direction of the arrows for all atoms we are tracking
+/datum/component/team_monitor/proc/update_all_directions()
+	if(!updating)
+		return
+	for(var/datum/component/tracking_beacon/beacon as() in tracking)
+		update_atom_dir(beacon)
+
+//Update the arrow towards another atom
+/datum/component/team_monitor/proc/update_atom_dir(datum/component/tracking_beacon/beacon)
+	if(!updating || !updating.hud_used || !beacon)
+		return
+	var/atom/movable/screen/arrow/screen = tracking[beacon]
+	var/turf/target_turf = get_turf(beacon.parent)
+	var/turf/parent_turf = get_turf(parent)
+	if(target_turf.z != parent_turf.z || target_turf == parent_turf)
+		if(screen)
+			//Remove the screen
+			if(updating.hud_used)
+				updating.hud_used.team_finder_arrows -= screen
+			qdel(screen)
+			tracking[beacon] = null
+			//Update their hud
+			if(updating.hud_used)
+				updating.hud_used.show_hud(updating.hud_used.hud_version, updating)
```

Comment:
```suggestion
			updating.hud_used.team_finder_arrows -= screen
			qdel(screen)
			tracking[beacon] = null
			//Update their hud
			updating.hud_used.show_hud(updating.hud_used.hud_version, updating)
```

Redundant checks; `updating.hud_used` is already checked at the beginning of this proc.